tests/unit/memefficiency.tcl

```diff
@@ -145,601 +287,323 @@ run_solo {defrag} {
-                $rd script load $val
-                $rd set k$j $val
-            }
-            for {set j 0} {$j < $n} {incr j} {
-                $rd read ; # Discard script load replies
-                $rd read ; # Discard set replies
-            }
-            after 1000 ;# give defrag some time to work
-            if {$::verbose} {
-                puts "used [s allocator_allocated]"
-                puts "rss [s allocator_active]"
-                puts "frag [s allocator_frag_ratio]"
-                puts "frag_bytes [s allocator_frag_bytes]"
-            }
-            assert_lessthan [s allocator_frag_ratio] 1.05
-
-            # Delete all the keys to create fragmentation
-            for {set j 0} {$j < $n} {incr j} { $rd del k$j }
-            for {set j 0} {$j < $n} {incr j} { $rd read } ; # Discard del replies
-            $rd close
-            after 120 ;# serverCron only updates the info once in 100ms
-            if {$::verbose} {
-                puts "used [s allocator_allocated]"
-                puts "rss [s allocator_active]"
-                puts "frag [s allocator_frag_ratio]"
-                puts "frag_bytes [s allocator_frag_bytes]"
-            }
-            assert_morethan [s allocator_frag_ratio] 1.4
-
-            catch {r config set activedefrag yes} e
-            if {[r config get activedefrag] eq "activedefrag yes"} {
-            
-                # wait for the active defrag to start working (decision once a second)
-                wait_for_condition 50 100 {
-                    [s total_active_defrag_time] ne 0
-                } else {
-                    after 120 ;# serverCron only updates the info once in 100ms
-                    puts [r info memory]
-                    puts [r info stats]
-                    puts [r memory malloc-stats]
-                    fail "defrag not started."
-                }
-
-                after 1000 ;# Give defrag time to work (might be multiple cycles)
+            perform_defrag_test $title {
+                # Populate memory with interleaving script-key pattern of same size
+                set dummy_script "--[string repeat x 400]\nreturn "
+                set rd [valkey_deferring_client]
+                for {set j 0} {$j < $n} {incr j} {
+                    set val "$dummy_script[format "%06d" $j]"
+                    $rd script load $val
+                    $rd set k$j $val
+                }
+                for {set j 0} {$j < $n} {incr j} {
+                    $rd read ; # Discard script load replies
+                    $rd read ; # Discard set replies
+                }
+            } fragment {
+                # Delete all the keys to create fragmentation
+                for {set j 0} {$j < $n} {incr j} { $rd del k$j }
+                for {set j 0} {$j < $n} {incr j} { $rd read } ; # Discard del replies
+                $rd close
+            } latency 100 ;# scripts aren't defragged incrementally, expect big latency
```

Comment:
Put this latency arg after the title but before all the code blocks? It'd be easier to spot it.